### ISSUE TITLE
Failed deployment of resources isn't reflected in API / Cosmos

### ISSUE THREAD

**Issue by tanya-borisova** [microsoft/AzureTRE#1783]
**Describe the bug**

When a resource deployment fails, the status in Cosmos still shows isActive = true, and the resource appears in a list of resources in the API. 

**Steps to reproduce**

I tried this on gitea workspace service and on shared services, but I think it would work on any resource 

1. Trivially break gitea workspace service, e.g. use a non-existing variable in Terraform
2. Build and publish gitea workspace service bundle
3. Run E2E tests and make sure they fail
4. The resource object in Cosmos will still have isActive set to true

**Comment by martinpeck** [comment]
Success/Failure is reflected in the Operations. The overall status isn't being accurately reflected in `isActive`,
This needs more thought.

**Comment by tanya-borisova** [comment]
@martinpeck  I still think this is a bug though, as I don't think we should be returning in API services that have failed to be deployed. `isActive` might not be the field for it, but in this case we can query operation results?

**Comment by martinpeck** [comment]
Agreed it's a bug (or, at very least, not an ideal situation). What's the next step for this? Does this need to be wrapped up in a conversation about how we convert operations into a headline status?

cc @damoodamoo @marrobi

**Comment by marrobi** [comment]
Agree, if initial deployment of a resource fails, shouldn't become active. If we get to updates, conversation might be different.

I do wonder if we need a resource state properly. So Queued, Deploying, Active, Updating, Disabled etc.

**Comment by tanya-borisova** [comment]
Discussed with David, plan is to use the new deploymentStatus field for the shared services logic (i.e.: don't deploy a new shared service, if a previous one has a successful deployment status). The field will be introduced here: https://github.com/microsoft/AzureTRE/pull/2001
We also think that this removed the need for having the isActive field, so we can remove it for simplicity
